spring-integration-ws/src/main/java/org/springframework/integration/ws/dsl/MarshallingWsOutboundGatewaySpec.java

```diff
@@ -0,0 +1,128 @@
+	 * @param destinationProvider the destination provider.
+	 * @param marshaller the marshaller.
+	 */
+	protected MarshallingWsOutboundGatewaySpec(DestinationProvider destinationProvider, Marshaller marshaller) {
+		this(destinationProvider, marshaller, null, null);
+	}
+
+	/**
+	 * Construct an instance with the provided arguments.
+	 * @param destinationProvider the destination provider.
+	 * @param marshaller the marshaller.
+	 * @param unmarshaller the unmarshaller.
+	 */
+	protected MarshallingWsOutboundGatewaySpec(DestinationProvider destinationProvider, Marshaller marshaller,
+			Unmarshaller unmarshaller) {
+		this(destinationProvider, marshaller, unmarshaller, null);
+	}
+
+	/**
+	 * Construct an instance with the provided arguments.
+	 * @param destinationProvider the destination provider.
+	 * @param marshaller the marshaller.
+	 * @param messageFactory the message factory.
+	 */
+	protected MarshallingWsOutboundGatewaySpec(DestinationProvider destinationProvider, Marshaller marshaller,
+			WebServiceMessageFactory messageFactory) {
+
+		this(destinationProvider, marshaller, null, messageFactory);
+	}
+
+	/**
+	 * Construct an instance with the provided arguments.
+	 * @param destinationProvider the destination provider.
+	 * @param marshaller the marshaller.
+	 * @param unmarshaller the unmarshaller.
+	 * @param messageFactory the message factory.
+	 */
+	protected MarshallingWsOutboundGatewaySpec(DestinationProvider destinationProvider,
+			@Nullable Marshaller marshaller, @Nullable Unmarshaller unmarshaller,
+			WebServiceMessageFactory messageFactory) {
+
+		this.target = new MarshallingWebServiceOutboundGateway(destinationProvider, marshaller, unmarshaller,
+				messageFactory);
+	}
+
+	/**
+	 * Construct an instance with the provided arguments.
+	 * @param uri the URI.
+	 * @param marshaller the marshaller.
+	 */
+	protected MarshallingWsOutboundGatewaySpec(String uri, Marshaller marshaller) {
+		this(uri, marshaller, (WebServiceMessageFactory) null);
+	}
+
+	/**
+	 * Construct an instance with the provided arguments.
+	 * @param uri the URI.
+	 * @param marshaller the marshaller.
+	 * @param messageFactory the message factory.
+	 */
+	protected MarshallingWsOutboundGatewaySpec(String uri, Marshaller marshaller,
+			WebServiceMessageFactory messageFactory) {
+
+		this(uri, marshaller, null, messageFactory);
+	}
+
+	/**
+	 * Construct an instance with the provided arguments.
+	 * @param uri the URI.
+	 * @param marshaller the marshaller.
+	 * @param unmarshaller the unmarshaller.
+	 */
+	protected MarshallingWsOutboundGatewaySpec(String uri, Marshaller marshaller, Unmarshaller unmarshaller) {
+		this(uri, marshaller, unmarshaller, null);
+	}
+
+	/**
+	 * Construct an instance with the provided arguments.
+	 * @param uri the URI.
+	 * @param marshaller the marshaller.
+	 * @param unmarshaller the unmarshaller.
+	 * @param messageFactory the message factory.
+	 */
+	protected MarshallingWsOutboundGatewaySpec(String uri,
```

Comment:
I think we need to consider to have all of these ctors spreaded as options and build the final `MarshallingWebServiceOutboundGateway` in the `doGet()`.
If possible, of course...

Otherwise it doesn't look too useful to have such a DSL spec based only on ctors

Reply:
Yeah - will rework - seemed a little clunky to me; I think a CTOR that takes a WST would be useful and return a different spec in that case (one that doesn't offer the properties that are disallowed with an explicit WST; there are many).